### PR TITLE
server: set master label before controllers

### PR DESCRIPTION
**Pull request by dweomer**
Set the master label and DNS config before applying controllers.

Also, fix the binary size check in CI.

### CODE REVIEW

File: scripts/binary_size_check.sh [hunk 1]
```diff
@@ -1,8 +1,9 @@
-#!/bin/sh
-
+#!/bin/bash
```
Comment (briandowns) [inline scripts/binary_size_check.sh:1]: Most systems have sh in some capacity, whether the bin or a symlink. I did this for compatibility purposes, especially given the latest issue that came through about nonbash compatibility with dash.

Comment (dweomer) [reply scripts/binary_size_check.sh:1]: While I generally laud portability, all of our scripts are bash already, except manifest.sh which is likely an oversight.

File: scripts/binary_size_check.sh [hunk 2]
```diff
@@ -1,8 +1,9 @@
...
 set -e
 
-if [ "${DEBUG}" = 1 ]; then
+cd $(dirname $0)/..
+
+if [ -n "${DEBUG}" ]; then
```
Comment (dweomer) [inline scripts/binary_size_check.sh:6]: @briandowns this test is why I switched to bash

**Review by brandond — APPROVED** [review]

**Comment by cjellick** [comment]
Is this for this issue? https://github.com/rancher/rke2/issues/127

**Comment by dweomer** [comment]
> Is this for this issue? [rancher/rke2#127](https://github.com/rancher/rke2/issues/127)

Yes, planned to add that after this was not longer draft. I notice i missed marking this as draft.

File: pkg/server/server.go [hunk 3]
```diff
@@ -142,11 +142,16 @@ func runControllers(ctx context.Context, config *Config) error {
 			panic(err)
 		}
 	}
+
 	if !config.DisableAgent {
-		go setMasterRoleLabel(ctx, sc.Core.Core().V1().Node())
+		if err := setMasterRoleLabel(ctx, sc.Core.Core().V1().Node()); err != nil {
```
[the remark this answers is not in the document]
Comment (dweomer) [reply pkg/server/server.go:147]: Yeah, i dunno. it doesn't appear to be deadlocking in either k3s or rke2 when I test it.

**Comment by cjellick** [comment]
no problem..im mainly just closing out browser tabs and making sure i have my Is crossed and Ts dotted

**Comment by brandond** [comment]
Looks good to me assuming there's no reason (other than perhaps reducing startup time) that we originally did it in a goroutine.

**Comment by brandond** [comment]
@dweomer do you want to pull the size check out to a separate PR? CI is currently logging a couple errors when trying to check the size:

```
ls: cannot access 'dist/artifacts/k3s': No such file or directory
1446 | sh: 61000000: unknown operand
```

**Pull request closed without merging**